pkg/build/android_linux.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package build
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/google/syzkaller/pkg/osutil"
+)
+
+// buildCuttlefishImage mounts a disk image, fetches and installs a Cuttlefish emulator binary,
+// and copies in the required kernel artifacts.
+func buildCuttlefishImage(params Params, bzImage, vmlinux, initramfs string) error {
+	tempDir, err := ioutil.TempDir("", "syz-build")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDir)
+	imageFile := filepath.Join(tempDir, "image")
+	if err := osutil.CopyFile(params.UserspaceDir, imageFile); err != nil {
+		return err
+	}
+	loop, loopFile, err := linuxSetupLoop(imageFile)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		unix.IoctlGetInt(loop, unix.LOOP_CLR_FD)
+		unix.Close(loop)
+	}()
+	mountDir := filepath.Join(tempDir, "mnt")
+	if err := osutil.MkdirAll(mountDir); err != nil {
+		return err
+	}
+	if err := tryMount(loopFile+"p1", mountDir); err != nil {
```

Comment:
Please dedup this with embedLinuxKernel, it copy-pastes lots of complex code.

One nice way to do this is to have a function mounts an image and calls a callback function:
```
func mountImage(imageFile string, cb func(imageDir string) error) {
   ...
   cb(tempDir)
   ...
}
```
Then linux and android can do own stuff while sharing the mounting code.